### PR TITLE
[Tables] Fixes Transactional Batch for insert merge/replace entity.

### PR DESCRIPTION
**Pull request by seankane-msft**
Fixes customer bug #16086

### CODE REVIEW

**Comment by richardpark-msft** [comment]
Might want to get in the habit of removing all the PR description boilerplate otherwise it looks like you didn't fill it out :)

**Comment by richardpark-msft** [comment]
Similarly, having a PR title with some more detail would be good as well. (still reviewing)

File: sdk/data/aztables/transactional_batch.go [hunk 1]
```diff
@@ -301,20 +301,24 @@ func (t *Client) generateEntitySubset(transactionAction *TransactionAction, writ
...
-		transactionAction.IfMatch = &star
-	}
+	// if transactionAction.IfMatch == nil {
+	// 	star := azcore.ETagAny
+	// 	transactionAction.IfMatch = &star
+	// }
 
 	switch transactionAction.ActionType {
 	case Delete:
+		ifMatch := "*"
```
Comment (richardpark-msft) [inline sdk/data/aztables/transactional_batch.go:311]: We have that fancy azcore.ETagAny constant for this, right?

File: sdk/data/aztables/transactional_batch.go [hunk 2]
```diff
@@ -301,20 +301,24 @@ func (t *Client) generateEntitySubset(transactionAction *TransactionAction, writ
 		return fmt.Errorf("entity properties must contain a %s property", rowKey)
 	}
 	// Consider empty ETags as '*'
-	if transactionAction.IfMatch == nil {
-		star := azcore.ETagAny
-		transactionAction.IfMatch = &star
-	}
+	// if transactionAction.IfMatch == nil {
```
Comment (RickWinter) [inline sdk/data/aztables/transactional_batch.go:304]: Remove the commented code

**Review by richardpark-msft — COMMENTED** [review]
Some questions.

File: sdk/data/aztables/client.go [hunk 3]
```diff
@@ -528,10 +528,26 @@ func (t *Client) UpdateEntity(ctx context.Context, entity []byte, options *Updat
...
 	case MergeEntity:
-		resp, err := t.client.MergeEntity(ctx, generated.Enum1Three0, t.name, partKey, rowkey, options.toGeneratedMergeEntity(mapEntity), &generated.QueryOptions{})
+		resp, err := t.client.MergeEntity(
+			ctx,
+			generated.Enum1Three0,
+			t.name,
+			partKey,
+			rowkey,
+			options.toGeneratedMergeEntity(mapEntity),
+			&generated.QueryOptions{},
```
[the remark this answers is not in the document]
Comment (seankane-msft) [reply sdk/data/aztables/client.go:538]: yes unfortunately you have to pass in a pointer to a defined struct.

File: sdk/data/aztables/transactional_batch.go (hunk 1 above)
Comment (seankane-msft) [reply sdk/data/aztables/transactional_batch.go:311]: Yes. The generated code takes a string, not the azcore.ETag type so this was slightly simpler.

File: sdk/data/aztables/transactional_batch.go [hunk 4]
```diff
@@ -346,7 +348,7 @@ func (t *Client) generateEntitySubset(transactionAction *TransactionAction, writ
 			entity[partitionKey].(string),
 			entity[rowKey].(string),
 			opts,
-			qo,
+			&generated.QueryOptions{},
```
Comment (chlowell) [inline sdk/data/aztables/transactional_batch.go:351]: Why does `InsertMerge` need different `QueryOptions`?

**Review by chlowell — APPROVED** [review]

File: sdk/data/aztables/transactional_batch.go (hunk 4 above)
Comment (seankane-msft) [reply sdk/data/aztables/transactional_batch.go:351]: The `qo` passed an `ETag` of `*`, which returns a ResourceNotFoundError if the entity does not already exist. Passing the empty struct will not send that header at all which creates the entity if it does not exist, which is the desired behavior.

**Pull request merged**